s3fs/core.py

```diff
@@ -120,6 +120,10 @@ async def _error_wrapper(func, *, args=(), kwargs=None, retries):
             err = e
             if "SlowDown" in str(e):
                 await asyncio.sleep(min(1.7**i * 0.1, 15))
+            elif "XAmzContentSHA256Mismatch" in str(e):
+                await asyncio.sleep(min(1.7**i * 0.1, 15))
+            elif "Bad Request" in str(e):
```

Comment:
Is there anything more specific about your case than Bad Request? I would have thought that the majority of those were not in fact retriable situations.

Reply:
The particular error line was `botocore.exceptions.ClientError: An error occurred (400) when calling the PutObject operation: Bad Request` , unfortunately I don't have the context anymore since the log was overwritten. But there was only one instance in the whole upload, and only required one retry before succeeding, which makes me think it was a transient connection error.

Reply:
OK to leave this one out, then, and assume it happens so rarely that others won't be impacted?

Reply:
sounds reasonable to me